subprojects/ide-native/src/main/java/org/gradle/ide/visualstudio/internal/NativeSpecVisualStudioTargetBinary.java

```diff
@@ -0,0 +1,299 @@
+        final String projectPath = binary.getComponent().getProjectPath();
+        if (":".equals(projectPath)) {
+            return ":" + taskName;
+        }
+
+        return projectPath + ":" + taskName;
+    }
+
+    @Override
+    public boolean isDebuggable() {
+        return !"release".equals(binary.getBuildType().getName());
+    }
+
+    @Override
+    public File getOutputFile() {
+        if (isExecutable()) {
+            InstallExecutable installTask = getInstallTask();
+            return new File(installTask.getInstallDirectory().get().getAsFile(), "lib/" + installTask.getSourceFile().get().getAsFile().getName());
+        } else {
+            return binary.getPrimaryOutput();
+        }
+    }
+
+    @Override
+    public List<String> getCompilerDefines() {
+        List<String> defines = new ArrayList<String>();
+        defines.addAll(getDefines("cCompiler"));
+        defines.addAll(getDefines("cppCompiler"));
+        defines.addAll(getDefines("rcCompiler"));
+        return defines;
+    }
+
+    private List<String> getDefines(String tool) {
+        PreprocessingTool rcCompiler = findCompiler(tool);
+        return rcCompiler == null ? Lists.<String>newArrayList() : new MacroArgsConverter().transform(rcCompiler.getMacros());
+    }
+
+    private PreprocessingTool findCompiler(String tool) {
+        return (PreprocessingTool) binary.getToolByName(tool);
+    }
+
+    @Override
+    public List<File> getIncludePaths() {
+        Set<File> includes = new LinkedHashSet<File>();
+
+        for (LanguageSourceSet sourceSet : binary.getInputs()) {
+            if (sourceSet instanceof HeaderExportingSourceSet) {
+                includes.addAll(((HeaderExportingSourceSet) sourceSet).getExportedHeaders().getSrcDirs());
+            }
+        }
+
+        for (NativeDependencySet lib : binary.getLibs()) {
+            includes.addAll(lib.getIncludeRoots().getFiles());
+        }
+
+        return new ArrayList<File>(includes);
+    }
+
+    @Override
+    public Iterable<VisualStudioTargetBinary> getDependencies() {
+        Set<VisualStudioTargetBinary> targetBinaries = Sets.newHashSet();
+        for (NativeLibraryBinary dependency : binary.getDependentBinaries()) {
+            if (dependency instanceof NativeBinarySpecInternal) {
+                targetBinaries.add(new NativeSpecVisualStudioTargetBinary((NativeBinarySpec) dependency));
+            }
+        }
+        return targetBinaries;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        NativeSpecVisualStudioTargetBinary that = (NativeSpecVisualStudioTargetBinary) o;
+
+        return binary.equals(that.binary);
+    }
+
+    @Override
+    public int hashCode() {
+        return binary.hashCode();
+    }
+
+    // TODO: There has to be a simpler way to do this.
```

Comment:
This is ugly looking, but if it works, it's probably OK to leave as-is.  I don't see anything obviously weird about it.

Reply:
It just seems overly complex, but I couldn't come up with something simpler given the relationship between `LanguageSourceSet`, its build dependencies and `SourceDirectorySet`.  We won't have to do this for the new plugins.